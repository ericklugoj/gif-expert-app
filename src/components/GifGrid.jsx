import React, { useState, useEffect } from 'react';
import GifGridItem from './GifGridItem';
import getGifs from '../helpers/getGifs';

const GifGrid = ({ category }) => {
	const [images, setImages] = useState([]);

	useEffect(() => {
		getGifs(category).then(setImages);
	}, [category]);

	return (
		<>
			<h3>{category}</h3>
			<div className='card-grid'>
				{images.map((img) => (
					<GifGridItem key={img.id} {...img} />
				))}
			</div>
		</>
	);
};

export default GifGrid;
