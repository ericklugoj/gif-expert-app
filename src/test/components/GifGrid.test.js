import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid/>', () => {
	const category = 'naruto';

	test('Debe de mostrarse correctamente', () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true,
		});

		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
		const gifs = [
			{
				id: 'ABC',
				url: 'https://localhost/test/test_image.jpg',
				title: 'test title',
			},
			{
				id: 'DEF',
				url: 'https://localhost/test/test_image2.jpg',
				title: 'test title 2',
			},
		];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});

		const wrapper = shallow(<GifGrid category={category} />);

		// expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('p').exists()).toBe(false);
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
	});
});
