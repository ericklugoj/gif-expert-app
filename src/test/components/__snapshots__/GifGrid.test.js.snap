// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounceIn"
  >
    naruto
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
